### DISCUSSION TITLE
Using CPU in Colab?

### DISCUSSION THREAD

**Discussion by ruidaiphd** [explosion/spaCy#9687] [Help: Model Advice] [answered]
Hello Guys,

This is actually a question. I am found a Colab NER code on line:
https://towardsdatascience.com/how-to-fine-tune-bert-transformer-with-spacy-3-6a90bfe57647

Everything seems ok, expect I always saw ℹ Using CPU in training, debugging and evaluating process. 

I am new to GPU version of spacy. Does it mean it is not using TPU on Colab?

Best,

Ray

**Comment by polm** [comment, marked as answer]
I'm not super familiar with Colab, but we have had reports of strange behavior with GPU related libraries on it, see [the FAQ](https://github.com/explosion/spaCy/discussions/8226#:~:text=GPU%20Issues%20on%20Google%20Colab). 

If you pass `-g 0` in the training step, it should either use the GPU or give an error if that's not possible.

To be clear so I can check this, can you provide a screenshot of where the "Using CPU" is displayed?

**Comment by ruidaiphd** [comment]
Thanks! It turns out that the refered method simply does not recognize Colab’s TPU. After I change it back to GPU accelerator. It works as expected.

Best,

Ray

Sent from my iPhone

On Nov 17, 2021, at 11:23 PM, polm ***@***.***> wrote:

﻿

I'm not super familiar with Colab, but we have had reports of strange behavior with GPU related libraries on it, see the FAQ<https://github.com/explosion/spaCy/discussions/8226#:~:text=GPU%20Issues%20on%20Google%20Colab>.

If you pass -g 0 in the training step, it should either use the GPU or give an error if that's not possible.

To be clear so I can check this, can you provide a screenshot of where the "Using CPU" is displayed?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/explosion/spaCy/discussions/9687#discussioncomment-1662212>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AHKVO3277IDDQFARZFZX7W3UMR5STANCNFSM5IHFO7ZQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by polm** [reply]
Glad you figured it out!
